### PR TITLE
INT-6770 - Additional vSphere feedback

### PR DESCRIPTION
**Pull request by adam-in-ict**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Summary

Additional updates based on user testing feedback.  Primarily, the session creation needed backward compatibility included since we won't yet know the API version we're querying.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,12 +35,12 @@
     "prepush": "yarn format:check && yarn lint && yarn type-check && jest --changedSince main"
   },
   "peerDependencies": {
-    "@jupiterone/integration-sdk-core": "^8.30.1"
+    "@jupiterone/integration-sdk-core": "^8.30.2"
```
Comment (adam-in-ict) [inline package.json:38]: Bumping to get access to API_VERSION as the step disabled reason.

File: src/client.ts [hunk 2]
```diff
@@ -149,10 +150,30 @@ export class APIClient {
         const sessionId = await res.json();
         this.sessionId = sessionId;
       } catch (err) {
+        this.logger.info(
+          { uri },
+          `Failed to authenticate.  Trying legacy session endpoint...`,
+        );
+      }
+      try {
```
Comment (adam-in-ict) [inline src/client.ts:158]: This is the main update.  We have to try the old endpoint if the main one fails, as we can't yet query the API version due to that endpoint requiring authentication.

File: src/steps/virtual-machine/index.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export async function fetchVms({
   if (guestQueryFailCount > 0) {
     logger.publishWarnEvent({
       name: IntegrationWarnEventName.MissingPermission,
-      description: `Could not query all guest information for VMs.  Success = ${guestQuerySuccessCount}  Failed = ${guestQueryFailCount}`,
+      description: `Could not query all guest information for VMs.  This may be due to permissions or having VMware tools turned off on those VMs.  Success = ${guestQuerySuccessCount}  Failed = ${guestQueryFailCount}`,
```
Comment (adam-in-ict) [inline src/steps/virtual-machine/index.ts:42]: Small info logging update based on past PR feedback.

File: src/client.ts [hunk 4]
```diff
@@ -149,10 +150,30 @@ export class APIClient {
...
+        // Next, try deprecated endpoint for backward compatibility
+        const res: Response = await fetch(uriDeprecated, {
```
Comment (VDubber) [inline src/client.ts:160]: nit: This block of code is duplicated from above. Future clean up could be to move this to a func and pass the different URLs.

**Review by VDubber — APPROVED** [review]

**Pull request merged**